From: Matthias Brugger <mbrugger@suse.com>
Subject: net: emac: Fix driver compilation
Patch-mainline: Never, fixes compilation for our kernel.
References: fate#320512

The kernel does not include the patches needed to support mdiobus_get_phy and 
phydev->mdio.dev. Change the driver to fix compilation errors.

Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
 drivers/net/ethernet/qualcomm/emac/emac-phy.c |    4 ++--
 drivers/net/ethernet/qualcomm/emac/emac.c     |    4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

--- a/drivers/net/ethernet/qualcomm/emac/emac-phy.c
+++ b/drivers/net/ethernet/qualcomm/emac/emac-phy.c
@@ -199,7 +199,7 @@ int emac_phy_config(struct platform_devi
 			 */
 			adpt->phydev = phy_find_first(mii_bus);
 		else
-			adpt->phydev = mdiobus_get_phy(mii_bus, phy_addr);
+			adpt->phydev = mii_bus->phy_map[phy_addr];
 
 		/* of_phy_find_device() claims a reference to the phydev,
 		 * so we do that here manually as well. When the driver
@@ -207,7 +207,7 @@ int emac_phy_config(struct platform_devi
 		 * without worrying about ACPI vs DT.
 		 */
 		if (adpt->phydev)
-			get_device(&adpt->phydev->mdio.dev);
+			get_device(&adpt->phydev->dev);
 	} else {
 		struct device_node *phy_np;
 
--- a/drivers/net/ethernet/qualcomm/emac/emac.c
+++ b/drivers/net/ethernet/qualcomm/emac/emac.c
@@ -732,7 +732,7 @@ static int emac_probe(struct platform_de
 err_undo_napi:
 	netif_napi_del(&adpt->rx_q.napi);
 err_undo_mdiobus:
-	put_device(&adpt->phydev->mdio.dev);
+	put_device(&adpt->phydev->dev);
 	mdiobus_unregister(adpt->mii_bus);
 err_undo_clocks:
 	emac_clks_teardown(adpt);
@@ -752,7 +752,7 @@ static int emac_remove(struct platform_d
 
 	emac_clks_teardown(adpt);
 
-	put_device(&adpt->phydev->mdio.dev);
+	put_device(&adpt->phydev->dev);
 	mdiobus_unregister(adpt->mii_bus);
 	free_netdev(netdev);
 
